Correction of Swimming Pool Evaporation Rate Bug and Documentation Errors

RKStrand opened the pull request:
## Pull request overview

This pull request fixes a problem with the evaporation rate calculation for indoor swimming pools as well as some documentation errors that were noted during the correction of the problem.  In addition, new output variables were added.  This corrects GitHub Issue #4852.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue #4852 as well as other issues uncovered during the work and looking at user input files.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes n/a
- [x] If new idf included, locally check the err file and other outputs
- [x] Documentation changes in place
- [x] Changed docs build successfully

RKStrand committed:
Code and input file changes

Changes to fix a bug and also increase the amount of available
information by the addition of new output variables.  The input file
was also modified to be at least a bit more realistic in certain
parameters.

RKStrand committed:
Merge branch 'develop' into 113836013-Swimming-Pool-Evaporation-and-Cover-Issues

RKStrand committed:
Unit test for bug fix

Created a unit test for evaporative rate, saturation pressure, and
partial pressure of air to confirm the corrected code is working

RKStrand committed:
Final docs additions and mods

Additions were made to the documentation for swimming pools as a result
of the defect fix.  In addition, some other documentation errors were
corrected as well.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 113836013-Swimming-Pool-Evaporation-and-Cover-Issues

Conflicts:
	testfiles/5ZoneSwimmingPool.idf

mjwitte commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-003/indoor-swimming-pool.tex:
```diff
@@ -59,22 +59,22 @@ \subsection{Convection from the pool water surface}\label{convection-from-the-po
 
 \begin{equation}
-h = 0.22 \cdot (T_p – T_a)1/3
+h = 0.22 \cdot (T_p - T_a)^{\sfrac{1}{3}}
```
This is showing as power 13, not 1/3.

RKStrand replied:
@mjwitte @Myoldmopar Okay...on this one, I have no idea what the problem is.  I got this to work at one point and then it changed later even though I had not changed it.  I followed this based on some information on-line and it did work the first time I built the docs after making this change.  However, after that, it didn't matter what I did on my computer it kept saying 13.  I thought maybe it was a glitch on my computer but apparently not.  What is the correct way to do this?

mjwitte commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-003/indoor-swimming-pool.tex:
```diff
@@ -141,6 +141,8 @@ \subsection{Evaporation from the pool water surface}\label{evaporation-from-the-
 
 When a cover is present, the cover and the cover evaporation factor reduce the amount of evaporation proportionally. For example, if the pool is half covered and the pool cover reduces convection by 50\%, the convective heat transfer coefficient is reduced by 25\% from the value calculated using the above equation. The value is converted to a latent gain (loss) through multiplication of the evapo ...
 
+The user should be aware of two key assumptions built into the equations for calculating the evaporation from the pool.  First, when the activity factor is zero, no evaporation will take place.  Thus, activity factor is not the same thing and should not be zero when there are no people in the pool as that will completely eliminate evaporation.  Second, when the cover evaporation factor is zero, t ...
```
. . . activity factor is not the same thing ??? as occupancy ???

RKStrand replied:
@mjwitte Yes, I know--it is not obvious and that is why I put that comment in there.  The Activity Factor term is based on a model from the literature.  Think of it as how active the pool surface is.  Even if the water is completely still, the activity factor can't be zero because it will cut off evaporation based on the model.  I was reminded of this while doing testing and thought that a comment in the documentation would avoid some future questions/problems.  If you feel that this needs more description, I am happy to add more here.

mjwitte replied:
The problem here is that the sentence is incomplete.  I was guessing that "as occupancy" should be inserted.

RKStrand replied:
@mjwitte Oic...now I get it.  Sorry, my brain was filling in the "as occupancy" without even realizing it and thought you were asking a different question.  Yes, I agree with your comment and will fix this when I fix the units issues.  Thanks for reviewing this!

mjwitte replied:
Sorry my proofreading marks weren't clear.  Get me some paper and a red pen so I can use proper editing marks. :-)

mjwitte commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-003/indoor-swimming-pool.tex:
```diff
@@ -152,38 +154,38 @@ \subsection{Conduction through the bottom of the pool}\label{conduction-through-
 where
 
-\(m_{fw}\) = Mass flow rate
+\(m_{fw}\) = Mass flow rate of fresh water (lb·s)
```
Shouldn't this be lb/s ?
And why is this whole section in IP units?  I'm not going to insist that it all be changed over now, but wondering why.

RKStrand replied:
@mjwitte yes, all of those should be lb/s.  I will correct all of them.  As to why there are IP units in this section, that's a good question.  I will need to go back and look at that.  Maybe the units are not needed.  I don't remember if there was something specific about the equations being from the literature and being developed in IP or if that was just a function of the original NFP having those units.  My preference would be to get rid of the units or have them be in SI.

mjwitte replied:
Units are always good, so I would not be in favor of dropping them.

mjwitte commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-003/indoor-swimming-pool.tex:
```diff
@@ -152,38 +154,38 @@ \subsection{Conduction through the bottom of the pool}\label{conduction-through-
 
 where
 
 \(m_{hw}\) = Mass flow rate (lb·s)
```
lb/s ?

mjwitte commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-003/indoor-swimming-pool.tex:
```diff
@@ -195,19 +197,19 @@ \subsection{Pool Heating to Control the Pool Water Temperature}\label{pool-heati
+\(T_{old}\) = temperature of water at the last time step (°F)
 
-\(m_p\) = needed mass flow rate of water from the plant
+\(m_p\) = needed mass flow rate of water from the plant (lb·s)
```
lb/s ?

mjwitte committed:
Swimming Pool minor doc fix and unit test return at end

mjwitte reviewed (changes requested):
@RKStrand I've merged in develop, resolved conflicts, and made a couple of minor corrections, so be sure to pull the branch before addressing comments.  

@Myoldmopar Strange, the CI build of the Engineering Reference was not complete (most of the document is missing), but it builds locally for me.  Will see how it comes out this time.

mjwitte committed:
SwimmingPool spacing perfection

RKStrand committed:
Docs changes as per review comments

Changed a variety of things in the engineering reference to address
comments brought up by @mjwitte during the review process.  This
version should address all of his comments.

RKStrand commented:
@mjwitte Ok, Mike, I believe that I have addressed all of your comments on the docs and have made another commit to reflect this.  Hopefully this takes care of the remaining issues for this issue, but if not let me know.  Thanks!

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 113836013-Swimming-Pool-Evaporation-and-Cover-Issues

mjwitte committed:
Fix engineering ref build problem

mjwitte commented:
@Myoldmopar Arrrrgh.  Bucket errors so the linux build didn't post any docs.  Deleting and trying again.  
@RKStrand Your changes are good.  This will merge once the unrelated doc changes I added are verified.

Myoldmopar commented:
I am reproducing this on the CI machine right now, let me fix it and push up a fix from there.  Please hold...

Myoldmopar committed:
TeX cleanups

Myoldmopar commented:
OK, that should do it, @mjwitte.  I was able to reproduce the build problem on the CI box, and then work through it, cleaning up bad math stuffs.  I don't mind if you just merge this in since I was able to build on the CI box, and who knows if it'll actually submit to Amazon anyway.

mjwitte commented:
@Myoldmopar Thanks.  Would be nice to add the PB/RB/CB macros to the documentation wiki page.  And maybe some links to the docs  for the packages we're using?  Something would explain, for example, how the various environments like `align*` are used.
